### PR TITLE
Updates reset password with govuk markup to make it look styled

### PR DESCRIPTION
**Pull request by paulmsmith**
The work to complete this ticket: https://trello.com/c/0SmwpZM0

### CODE REVIEW

File: app/views/devise/passwords/edit.html.erb [hunk 1]
```diff
@@ -6,28 +6,29 @@
          }
 %>
 
-<h2>Change your password</h2>
+<h2 class="govuk-heading-l">Change your password</h2>
```
Comment (stuartmccoll) [inline app/views/devise/passwords/edit.html.erb]:
```suggestion
<h1 class="govuk-heading-l">Change your password</h1>
```

Probably just a hangover from the original code, but guessing this should actually be a `h1`?

File: app/views/devise/passwords/edit.html.erb [hunk 2]
```diff
@@ -6,28 +6,29 @@
...
-  <div class="actions">
-    <%= f.submit "Change my password" %>
+  <div class="govuk-form-group">
+    <%= render(ButtonComponent.new(element: "input", text: "Change my password")) %>
   </div>
-<% end %>
 
-<%= render "devise/shared/links" %>
+  <%- if devise_mapping.confirmable? && controller_name != 'confirmations' %>
+    <%= link_to "Didn't receive confirmation instructions?", new_confirmation_path(resource_name) %><br />
```
Comment (gidsg) [inline app/views/devise/passwords/edit.html.erb]: what is this `<br />` doing. Could we remove it?

File: app/views/devise/passwords/edit.html.erb [hunk 3]
```diff
@@ -6,28 +6,29 @@
...
 
-  <div class="field">
-    <%= f.label :password, "New password" %><br />
-    <% if @minimum_password_length %>
-      <em>(<%= @minimum_password_length %> characters minimum)</em><br />
-    <% end %>
-    <%= f.password_field :password, autofocus: true, autocomplete: "new-password" %>
+  <div class="govuk-form-group">
+    <%= f.label :password, "New password", class: "govuk-label" %>
+    <div class="govuk-hint"><% if @minimum_password_length %>Must be at least <%= @minimum_password_length %> characters<% end %></div>
```
Comment (stuartmccoll) [inline app/views/devise/passwords/edit.html.erb]: Does it make more sense here to wrap the `div` inside the `if` condition? If `@minimum_password_length` was somehow not set, we'd end up rendering an empty `div`.

```suggestion
    <% if @minimum_password_length %><div class="govuk-hint">Must be at least <%= @minimum_password_length %> characters</div><% end %>
```

File: app/views/devise/passwords/edit.html.erb (hunk 2 above)
Comment (paulmsmith) [reply app/views/devise/passwords/edit.html.erb]: Good spot. Done.

File: app/views/devise/passwords/edit.html.erb (hunk 1 above)
Comment (stuartmccoll) [reply app/views/devise/passwords/edit.html.erb]: I think you resolved this but didn't commit the suggestion - unresolving it, but shout at me if that was intentional 😂!

**Pull request merged**